examples/example-shader/src/lib.rs

```diff
@@ -160,29 +151,37 @@ pub fn main_fs(input: Input<Vec4>, mut output: Output<Vec4>) {
         w_axis: Vec4(0.0, -0.14834046, -0.98893654, 0.0),
     };
 
-    let cs_pos = Vec4(dir.0, -dir.1, 1.0, 1.0);
-    let ws_pos = {
-        let p = clip_to_world.mul_vec4(cs_pos);
+    let clip_pos = in_pos.load();
+    let world_pos = {
+        let p = clip_to_world.mul_vec4(clip_pos);
         p.truncate() / p.3
     };
-    let dir = (ws_pos - eye_pos).normalize();
+    let dir = (world_pos - eye_pos).normalize();
 
     // evaluate Preetham sky model
     let color = sky(dir, sun_pos);
 
-    output.store(color.extend(0.0))
+    output.store(color.extend(1.0))
 }
 
 #[allow(unused_attributes)]
 #[spirv(entry = "vertex")]
 pub fn main_vs(
-    in_pos: Input<Vec4>,
-    _in_color: Input<Vec4>,
-    #[spirv(builtin = "position")] mut out_pos: Output<Vec4>,
-    mut out_color: Output<Vec4>,
+    #[spirv(builtin = "vertex_index")] vert_idx: Input<i32>,
+    #[spirv(builtin = "position")] mut gl_pos: Output<Vec4>,
+    mut out_pos: Output<Vec4>,
```

Comment:
Could we call these `out_pos` and `out_pos_interpolator` or something? I would love us to not have the  `gl_` prefix anywhere if we can avoid it.